cmd/juju/cloud/add.go

```diff
@@ -192,6 +195,14 @@ func (c *AddCloudCommand) runInteractive(ctxt *cmd.Context) error {
 		return errors.Trace(err)
 	}
 
+	// At this stage, since we do not have a reference to any model, nor can we get it,
+	// nor do we need to have a model for anything that this command does,
+	// no cloud credential stored server-side can be invalidated.
+	// So, just log an informative message.
+	c.CloudCallCtx.InvalidateCredentialF = func(reason string) error {
+		ctxt.Infof("Cloud credential is not accepted by cloud provider: %v", reason)
```

Comment:
Should this be a Warningf?

Reply:
Infof on command context goes to Stderr. So, no Infof here is better suited.